ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -327,11 +327,16 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 			} else {
 				// We only need to delete the RayCluster. We don't need to delete the submitter Kubernetes Job so that users can still access
 				// the driver logs. In addition, a completed Kubernetes Job does not actually use any compute resources.
+
 				if _, err = r.deleteClusterResources(ctx, rayJobInstance); err != nil {
 					return ctrl.Result{RequeueAfter: RayJobDefaultRequeueDuration}, err
 				}
+				if err = r.Client.Delete(ctx, rayJobInstance); err != nil {
```

Comment:
If we delete a RayJob, its associated RayCluster will also be deleted automatically. We do not need to delete the RayCluster before deleting the RayJob.

Reply:
Since we are now deleting RayJobs based on `ttlSecondsAfterFinished`, should it be possible to set `ttlSecondsAfterFinished` with `shutdownAfterJobFinishes = false`? I think in the current state, `ttlSecondsAfterFinished` only applies when `shutdownAfterJobFinishes = true`.

Something like:
* shutdownAfterJobFinishes=true, ttlSecondsAfterFinished > 0: delete the cluster right away, keep the RayJob until `ttlSecondsAfterFinished`. 
* shutdownAfterJobFinishes=false, ttlSecondsAfterFinished > 0: don't delete the cluster right away, delete the RayJob *and cluster) after ttlSecondsAfterFinished
* shutdownAfterjobFinishes=false|true, ttlSecondsAfterFinished=0: delete the RayJob right away

Reply:
On second thought, this could be a breaking change, because today `shutdownAfterJobFinishes=true, ttlSecondsAfterFinished > 0` means delete the RayCluster after `ttlSecondsAfterFinished`

Reply:
Thanks for @kevin85421 for reminding me🙏

 [KuberRay Doc](https://github.com/ray-project/kuberay) refer that "You can also configure RayJob to automatically delete the RayCluster once the job finishes."
So, is the default configuration for RayJob to delete the RayCluster?
How can I find code about this?

Reply:
It's not obvious, but because we don't specifty a default override for `shutdownAfterJobFinishes`, it uses the bool value of false for the field https://github.com/ray-project/kuberay/blob/3c44ba0131ff94e69db62418bdace72df0e24dbb/ray-operator/apis/ray/v1/rayjob_types.go#L79-L80